src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -444,4 +444,47 @@ public PortalSpawnEvent(IWorld world, BlockPos pos, BlockState state, NetherPort
             return size;
         }
     }
+    
+    /**
+     * Fired when when this block is right clicked by a tool to change its state.
+     * Used to determine if an axe can strip, a shovel can path, or a hoe can till.
+     * 
+     * This event is {@link Cancelable}. If canceled, this will prevent the tool
+     * from changing the block's state.
+     */
+    @Cancelable
+    public static class BlockToolInteractEvent extends BlockEvent
+    {
+
+    	private final PlayerEntity player;
+    	private final ItemStack stack;
+    	private BlockState finalState;
+    	
+		public BlockToolInteractEvent(IWorld world, BlockPos pos, BlockState originalState, PlayerEntity player, ItemStack stack) {
+			super(world, pos, originalState);
+			this.player = player;
+			this.stack = stack;
+			this.finalState = originalState;
```

Comment:
There really should be a private final originalState, so there is always a clean reference to the originalState.

Reply:
I extended BlockEvent, which holds a private final originalState.

Reply:
Ah fair enough, ignore me :P

Reply:
You can resolve this